src/olympia/shelves/admin.py

```diff
@@ -1,9 +1,30 @@
-from django.contrib import admin
+from django.contrib import admin, messages
+from django.db.models import F
 
-from .forms import ShelfForm
+from .forms import ShelfForm, ShelfManagementForm
 
 
 class ShelfAdmin(admin.ModelAdmin):
     list_display = ('title', 'shelf_type')
     actions = ['delete_selected']
     form = ShelfForm
+
+
+class ShelfManagementAdmin(admin.ModelAdmin):
+    list_display = ('position', 'shelf', 'enabled')
+    list_display_links = ('shelf',)
+    list_editable = ('position', 'enabled')
+    actions = ['delete_selected']
+    ordering = [F('position').asc(nulls_last=True)]
+    form = ShelfManagementForm
+
+    def save_model(self, request, obj, form, change):
```

Comment:
Mmm, I'm not a fan of defining this validation twice.  I suggest moving the `def clean` from the form to `ShelfManagement` model.  (You probably don't need `ShelfManagementForm` at that point either - just define `fields` on `ShelfManagementAdmin`)